tests/test.bats

```diff
@@ -6,61 +14,106 @@
 
 setup() {
   set -eu -o pipefail
-  TEST_BREW_PREFIX="$(brew --prefix)"
-  load "${TEST_BREW_PREFIX}/lib/bats-support/load.bash"
-  load "${TEST_BREW_PREFIX}/lib/bats-assert/load.bash"
 
+  # Override this variable for your add-on:
+  export GITHUB_REPO=rfay/ddev-php-patch-build
+
+  TEST_BREW_PREFIX="$(brew --prefix 2>/dev/null || true)"
+  export BATS_LIB_PATH="${BATS_LIB_PATH}:${TEST_BREW_PREFIX}/lib:/usr/lib/bats"
+  bats_load_library bats-assert
+  bats_load_library bats-file
+  bats_load_library bats-support
 
-  export DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )/.."
-  export TESTDIR=~/tmp/ddev-php-patch-build-test
-  mkdir -p $TESTDIR
-  export PROJNAME=test-php-patch-build
-  export DDEV_NON_INTERACTIVE=true
-  ddev delete -Oy ${PROJNAME} >/dev/null 2>&1 || true
+  export DIR="$(cd "$(dirname "${BATS_TEST_FILENAME}")/.." >/dev/null 2>&1 && pwd)"
+  export PROJNAME="test-$(basename "${GITHUB_REPO}")"
+  mkdir -p "${HOME}/tmp"
+  export TESTDIR="$(mktemp -d "${HOME}/tmp/${PROJNAME}.XXXXXX")"
+  export DDEV_NONINTERACTIVE=true
+  export DDEV_NO_INSTRUMENTATION=true
+  ddev delete -Oy "${PROJNAME}" >/dev/null 2>&1 || true
   cd "${TESTDIR}"
-  ddev config --project-name=${PROJNAME} --fail-on-hook-fail --php-version=8.4
-  ddev start -y
-  ddev describe
+  run ddev config --project-name="${PROJNAME}" --project-tld=ddev.site
+  assert_success
+  run ddev start -y
+  assert_success
+
   export CUSTOM_PHP_MINOR_VERSION="8.2.23"
-  cat <<EOF >index.php
-<?php
-echo phpversion();
-EOF
+  printf "<?php\necho phpversion();\n" >index.php
+  assert_file_exist index.php
 }
 
 health_checks() {
   run ddev php --version
   assert_success
   assert_output --partial "PHP ${CUSTOM_PHP_MINOR_VERSION}"
-  run curl --fail -s https://${PROJNAME}.ddev.site/
+
+  run curl -sf https://${PROJNAME}.ddev.site/
   assert_success
   assert_output --partial "${CUSTOM_PHP_MINOR_VERSION}"
+
+  run ddev describe
+  assert_success
+  assert_output --partial "Static PHP: ${CUSTOM_PHP_MINOR_VERSION}"
+
+  run ddev php -m
+  assert_success
+  assert_output --partial "apcu"
+  assert_output --partial "bcmath"
+  assert_output --partial "ctype"
+  assert_output --partial "curl"
+  assert_output --partial "dom"
+  assert_output --partial "fileinfo"
+  assert_output --partial "filter"
+  assert_output --partial "ftp"
+  assert_output --partial "gd"
+  assert_output --partial "mbstring"
+  assert_output --partial "openssl"
+  assert_output --partial "pdo_mysql"
+  assert_output --partial "pdo_pgsql"
+  assert_output --partial "pdo_sqlite"
+  assert_output --partial "session"
+  assert_output --partial "sqlite3"
+  assert_output --partial "tokenizer"
+  assert_output --partial "xml"
```

Comment:
This is fine and simple, but could be done prettier with a for loop. It could also be done with a regex listing all of them.